src/viam/rpc/dial.py

```diff
@@ -30,7 +30,7 @@ class Credentials:
     Currently only supports robot location secret.
     """
 
-    type: Literal["robot-location-secret"]
+    type: str
```

Comment:
If the number of auth types starts growing significantly we can revisit, but for now I think we still want this to be a `Literal` (don't want to tell people they can pass _anything_ if we weren't previously). But, instead of just the one, we should instead have a `Union` of `robot-location-secret` and `robot-secret` literals.